using System;
namespace customException
{
    public class Program
    {
        static void Main(string[] args)
        {
            try
            {
                int result = CheckAge();
                Console.WriteLine(result);
            }
            catch (CustomException ex)  
            {
                Console.WriteLine(ex.Message);  
            }
            catch (Exception ex)  
            {
                Console.WriteLine(ex);
            }


        }
        static int CheckAge()
        {
            Console.WriteLine("Enter Age:");
            int age = Convert.ToInt32(Console.ReadLine());
            if (age < 18)
            {
                throw new CustomException("Age should be above 18");
            }
            return age;


        }
        public class CustomException : Exception
        {
            public CustomException(string message) : base(message)
            {
            }
        }

    }
}
